### ISSUE TITLE
E0202: (method-hidden) false positive when mixed member/method

### ISSUE THREAD

**Issue by socketpair** [PyCQA/pylint#3936]
Generates false positive:
```python
class Parent:
    def __init__(self):
        self.__check_should_start = None

class Child(Parent):
    def __check_should_start(self):
        pass
```

works OK:
```python
class Parent:
    def __check_should_start(self):
        pass

class Child(Parent):
    def __check_should_start(self):
        pass
```

```
$ pylint-3 --version
pylint 2.4.4
astroid 2.3.3
Python 3.8.6 (default, Sep 25 2020, 00:00:00) 
[GCC 10.2.1 20200723 (Red Hat 10.2.1-1)]
```

**Comment by hippo91** [comment]
@socketpair thanks for the report.
However it don't seem to be a false positive:
```
pylint --help-msg method-hidden
:method-hidden (E0202): *An attribute defined in %s line %s hides this method*
  Used when a class defines a method which is hidden by an instance attribute
  from an ancestor class or set by some client code. This message belongs to
  the classes checker.
```
It seems that the help message correspond exactly to the situation.
I close this issue but do not hesitate to answer if you think i am wrong. I wiil reopen the issue if it is the case.

**Comment by socketpair** [comment]
NO! It does not hide ! Since  they are PRIVATE variables. And of course they are different. If it were `._example` (i.e. protected var) it would be OK to raise the warning.

**Comment by hippo91** [comment]
@socketpair thanks. I seems a bug then.

**Comment by hippo91** [comment]
For anyone wanting to contribute, please have a look to the `classes.py` module in the method `visit_functiondef`.
